### PR TITLE
chore: bump mdxjs dependencies

### PR DESCRIPTION
**Pull request by atanasster**
issue: #1345 

Bump versions @mdx-js/mdx and @mdx-js/react to "^1.6.22", gatsby-plugin-mdx to "^1.6.0"

### CODE REVIEW

**Review by lachlanjc — APPROVED** [review]

**Comment by 454de6e** [comment]
See https://github.com/system-ui/theme-ui/issues/1345#issuecomment-742225675.

**Comment by hasparus** [comment]
Pasting @454de6e's comment.

> When using theme-ui in an MDX project (which of course is optional because theme-ui can be used without any MDX), I would expect you would already bring your own @mdx-js/react as peer dependency of whatever framework you use, e.g. gatsby-plugin-mdx. In that case the hard-wired dependency is in the way.
> 
> But I also see your point that you want theme-ui usable without the hassle of having to install peer dependencies. So I wonder if maybe Theme UI should also have an official way of using it modularly, where rather than installing theme-ui you reach for specific packages like @theme-ui/components or @theme-ui/mdx (complemented by @theme-ui/core) depending on your specific styling needs. Then @theme-ui/mdx can have @mdx-js/react as peer dependency and whoever wants the complete experience can install theme-ui which comes with @mdx-js/react as dependency.

We can merge this, but let's make `@mdx-js/react` a peer dep in `theme-ui` and `@theme-ui/mdx`, and align the docs with it.

1. We minimize our surface our surface a bit. Right now Theme UI seems like a huge colossus which does everything. If it's a dependency, people don't have to read its docs, right? And then we have problems.
2. It exports a context provider. Any version mismatch here is breaking the _provider-consumer_ link, so the entire thing stops working.

I have mdx-js versions locked in resolutions in all of my Theme UI + MDX projects. It works but it isn't an experience we'd like the users to have, especially since you can't do it with `npm` AFAIK.

**Review by hasparus — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## v0.6.0 UNRELEASED
 
+- Bump versions @mdx-js/mdx and @mdx-js/react to "^1.6.22", gatsby-plugin-mdx to "^1.6.0"
```
Comment (lachlanjc) [inline CHANGELOG.md]:
```suggestion
- Bump versions `@mdx-js/mdx` and `@mdx-js/react` to `^1.6.22`, gatsby-plugin-mdx to `^1.6.0`. PR #1351
```

**Review by lachlanjc — APPROVED** [review]

**Comment by atanasster** [comment]
> Pasting @454de6e's comment.
> 
> > When using theme-ui in an MDX project (which of course is optional because theme-ui can be used without any MDX), I would expect you would already bring your own @mdx-js/react as peer dependency of whatever framework you use, e.g. gatsby-plugin-mdx. In that case the hard-wired dependency is in the way.
> > But I also see your point that you want theme-ui usable without the hassle of having to install peer dependencies. So I wonder if maybe Theme UI should also have an official way of using it modularly, where rather than installing theme-ui you reach for specific packages like @theme-ui/components or @theme-ui/mdx (complemented by @theme-ui/core) depending on your specific styling needs. Then @theme-ui/mdx can have @mdx-js/react as peer dependency and whoever wants the complete experience can install theme-ui which comes with @mdx-js/react as dependency.
> 
> Let's merge this, but let's make `@mdx-js/react` a peer dep in `theme-ui` and `@theme-ui/mdx`, and align the docs with it.
> 
> 1. We minimize our surface our surface a bit. Right now Theme UI seems like a huge colossus which does everything. If it's a dependency, people don't have to read its docs, right? And then we have problems.
> 2. It exports a context provider. Any version mismatch here is breaking the _provider-consumer_ link, so the entire thing stops working.
> 
> I have mdx-js versions locked in resolutions in all of my Theme UI + MDX projects. It works but it isn't an experience we'd like the users to have, especially since you can't do it with `npm` AFAIK.

I believe also just installing a newer version of mdxjs should solve, not just using resolutions. Can you pls confirm?

**Comment by hasparus** [comment]
> I believe also just installing a newer version of mdxjs should solve, not just using resolutions. Can you pls confirm?

Yup, it's still a problem for me. Regardless of the version, we can still get conflicts.

**Comment by hasparus** [comment]
Let's follow up in #1388.

**Pull request merged**